kedro/framework/cli/starters.py

```diff
@@ -149,11 +217,32 @@ def starter():
 @starter.command("list")
 def list_starters():
     """List all official project starters available."""
-    repo_url = _STARTERS_REPO.replace("git+", "").replace(".git", "/tree/main/{alias}")
-    output = [
-        {alias: repo_url.format(alias=alias)} for alias in sorted(_STARTER_ALIASES)
-    ]
-    click.echo(yaml.safe_dump(output))
+    starters_aliases = _get_starters_aliases()
+    # we "reverse" the dictionnary so that the origin key in starter config
+    # {name: {template_path; ..., origin: ...}} became the first key:
+    # {kedro: ["astro-iris":{...}], "my-plugin":["my-starter":{...}]}
+    starters_by_origin = {}
+    for starter_name, starter_config in starters_aliases.items():
+        starter_raw_config = {
+            starter_name: {k: v for k, v in starter_config.items() if k != "origin"}
+        }
+        if starter_config["origin"] in starters_by_origin:
+            starters_by_origin[starter_config["origin"]].append(starter_raw_config)
+        else:
+            starters_by_origin[starter_config["origin"]] = [starter_raw_config]
+    for origin, starter_aliases in starters_by_origin.items():
+        click.secho(f"Starters from '{origin}'\n", fg="yellow")
+        output = []
+        for starter_alias in starter_aliases:
+            for starter_name, starter_config in starter_alias.items():
+                starter_name_displayed = (
+                    f"{starter_name} "
+                    f"(template_path={starter_config.get('template_path')}, "
+                    f"directory={starter_config.get('directory')}, "
+                    f"checkout={starter_config.get('checkout')})"
+                )
+                output.append(starter_name_displayed)
```

Comment:
I've made another attempt with ``yaml.dump``. i am a bit worried about readibilty though, since it displayed inforamtion on several lines for each starter and there are a lot of them, so it may hurt discoverability. Current implementation looks like this: 

![image](https://user-images.githubusercontent.com/29451317/156063326-83181bf1-1a2c-406e-823c-15bfcd7faad4.png)
 I can reorder the key (to have "name" as the first key) a d remove unused key (``origin`` or ``directory`` when it is ``None``), but I still find it harder to read than above suggestion. What do you think?